src/features/comments/commentsSlice.ts

```diff
@@ -0,0 +1,65 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getPostComments, createComment, deleteComment } from '../../api/comments';
+import { Comment } from '../../types/Comment';
+
+export interface CommentsState {
+  items: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: CommentsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+}
+
+export const fetchComments = createAsyncThunk(
+  'comments/fetchComments',
+  async (postId: number) => {
+  return await getPostComments(postId);
+});
+
+export const addComment = createAsyncThunk(
+  'comments/addComment',
+  async (data: Omit<Comment, 'id'>) => {
+    return await createComment(data);
+  }
+);
+
+export const removeComment = createAsyncThunk(
+  'comments/removeComment',
+  async (commentId: number) => {
+    await deleteComment(commentId);
+    return commentId;
+  }
+);
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchComments.pending, (state) => {
+        state.loaded = false;
+        state.hasError = false;
+      })
+      .addCase(fetchComments.fulfilled, (state, action) => {
+        state.loaded = true;
+        state.items = action.payload;
+      })
+      .addCase(fetchComments.rejected, (state) => {
+        state.loaded = true;
+        state.hasError = true;
```

Comment:
Consider adding more detailed error handling or logging to provide better insights into what went wrong during the fetch operation.